lib/Driver/Driver.cpp

```diff
@@ -1755,8 +1757,17 @@ Driver::computeCompilerMode(const DerivedArgList &Args,
     return Inputs.empty() ? OutputInfo::Mode::REPL
                           : OutputInfo::Mode::Immediate;
 
-  const Arg *ArgRequiringSingleCompile = Args.getLastArg(
-      options::OPT_whole_module_optimization, options::OPT_index_file);
+  bool HasWMO = Args.hasFlag(options::OPT_whole_module_optimization,
+                             options::OPT_no_whole_module_optimization,
+                             false);
+
+  Arg *ArgRequiringSingleCompile = nullptr;
+  if (HasWMO) {
+    ArgRequiringSingleCompile = Args.getLastArg(options::OPT_index_file,
+                                                options::OPT_whole_module_optimization);
+  } else {
+    ArgRequiringSingleCompile = Args.getLastArg(options::OPT_index_file);
```

Comment:
If you want to golf this, I'm guessing the following should work.

```
Arg *ArgRequiringSingleCompile = Args.getLastArg(options::OPT_index_file, HasWMO ? options::OPT_whole_module_optimization : llvm::OptSpecifier());
```

Reply:
Changed to that. I spent a while trying to find a `getLastArg` that would handle the negation too, but alas